assets/scss/_search.scss

```diff
@@ -1,165 +1,149 @@
 .search {
-  .search__input {
-    font-size: 14px;
-    border-radius: 3px;
-    border: 1px solid $grey-bg;
-    box-sizing: border-box;
-    padding: 10px;
-    padding-right: 35px;
     width: 100%;
-  }
-
-  &.search--sidebar {
+    max-width: calc(100% - 220px);
     position: relative;
-    width: 100%;
+    height: 100%;
+    font-size: 18px;
+    .search__input {
+        font-size: 18px;
+        border: none;
+        border-right: 1px solid $grey-bg;
+        box-sizing: border-box;
+        padding: 10px 76px;
+        height: 100%;
+        width: 100%;
+        &::placeholder {
+            font-size: 18px;
+        }
+        @media (max-width: 1000px) {
+            border-left: 1px solid $grey-bg;
+        }
+    }
     @media (max-width: 1000px) {
-      display: none;
+        flex: 1;
+        &--header {
+            display: none;
+        }
+    }
+    &--sidebar {
+        z-index: 1;
+        width: 100%;
+        max-width: unset;
+        display: none;
+        @media (max-width: 1000px) {
+            display: block;
+        }
     }
-    margin: 10px 0;
-    width: 100%;
-  }
 
-  &.search--header {
-    position: absolute;
-    right: 10px;
-    width: 150px;
-    transition: width 150ms ease-in-out;
-    z-index: 1;
+    .search__icon {
```

Comment:
Either add `pointer-events: none` and for empty input `pointer: cursor` or add onclick handler for search icon that would focus search input.